scripts/scr_get_item_names/scr_get_item_names.gml

```diff
@@ -0,0 +1,806 @@
+/// @param {bool} _show_available_only - Whether to limit the selection to what is in inventory, or show all items.
+/// @param {bool} _master_crafted_only - Whether to show only master crafted items, or hide them.
+/// @returns {array} The list of items to populate the selection list with.
+function scr_get_item_names(_equipment_type, _equipment_subtype, _unit_type, _include_company_standard=false, _show_available_only=false, _master_crafted_only=false) {
+    show_debug_message("scr_get_item_names: " + string(_equipment_type) + ", " + string(_equipment_subtype) + ", " + string(_unit_type) + ", " + string(_include_company_standard) + ", " + string(_show_available_only) + ", " + string(_master_crafted_only));
+    var _item_names;
+    switch(_unit_type) {
+        case eUNIT_TYPE.Infantry:
+            switch (_equipment_type) {
+                case eEQUIPMENT_TYPE.PrimaryWeapon:
+                case eEQUIPMENT_TYPE.SecondaryWeapon:
+                    if (_equipment_subtype == eEQUIPMENT_SUBTYPE.Ranged) {
+                        if (_show_available_only) {
+                            _item_names = get_filtered_equipment_item_names(
+                                "weapon",
+                                false, // ranged
+                                _master_crafted_only,
+                                undefined, // no required tags
+                                ["vehicle"], // exclude vehicle weapons
+                                true, // with_none
+                                true  // with_any
+                            );
+                        } else {
+                            _item_names = get_none_or_any_item_names(true, false);
+                            _item_names = push_marine_ranged_weapons_item_names(_item_names);
+                        }
+                    } else if (_equipment_subtype == eEQUIPMENT_SUBTYPE.Melee) {
+                        if (_show_available_only) {
+                            _item_names = get_filtered_equipment_item_names(
+                                "weapon",
+                                true, // melee
+                                _master_crafted_only,
+                                undefined, // no required tags
+                                ["vehicle"], // exclude vehicle weapons
+                                true, // with_none
+                                true // with_any
+                            );
+                            if (_include_company_standard) {
+                                array_push(_item_names, "Company Standard");
+                            }
+                        } else {
+                            _item_names = get_none_or_any_item_names(true, false);
+                            _item_names = push_marine_melee_weapons_item_names(_item_names);
+                        }
+                    } else {
+                        show_error("scr_get_item_names: Invalid equipment subtype for infantry", true);
```

Comment:
Is it intended to abort the game in case this fails? I understand that this is a last resort scenario and don't suggest wrapping this into `try_and_report` or just try-catch, but is it impossible to just not abort it? I assume if it gets bad, it'll crash anyway, but force crashing it is a bit, eh.

Reply:
I was assuming it was better to crash early and often. At my job, I want the develop who did something to pass the wrong args knows it went wrong or so if users encounter issues they can get reported. I realize in hindsight that doesn't necessarily apply to CM, if just one dev has that attitude.

What would you do in this situation?

Reply:
> I want the developer who passed the wrong arguments to the function to know it went wrong, or for issues users encounter to get reported.
> 
> What would you do in this situation?

Well, this is true. I'm just afraid of these things slipping and then causing crashes, when they could've just caused an error popup. I think crash is just more nasty than an error, and I'm more in preference of soft error throwing and not intentional crashing.
If anything, I would prefer godot/unity way of wrapping all user code in the project into try-catch and continuing to run the game in all cases, if the error is not critical. It just makes the game more robust, even if we're not around.

Reply:
> In [3e60f57](https://github.com/EttyKitty/ChapterMaster/commit/3e60f5713469dfff41df470120ea976b912826ec), I've switched to triggering an error popup. Let me know what you think.

Yeah, that works. I wanted to separate exception handling and error display as well, good idea.
But it'll probably now clash with https://github.com/EttyKitty/ChapterMaster/commit/b2a0d5015ce193dc40e72c7c2b77e36b64572de7.
So we're in for a fun conflict solving.

Reply:
sure, you know what you were up to and reviewed what I was up to.